internal/route/router.go

```diff
@@ -155,7 +142,7 @@ func (r *SimpleRouter) Route(
 		strconv.FormatBool(cached),
 	).Inc()
 
-	if err != nil {
+	if err != nil || httpResponse.StatusCode > http.StatusBadRequest {
```

Comment:
This fixes an issue where the HTTP response codes aren't added to the `node_gateway_router_upstream_rpc_request_errors.count` metric
![Uploading Screenshot 2023-12-18 at 6.29.57 PM.png…]()

Reply:
![Screenshot 2023-12-18 at 6 29 57 PM](https://github.com/satsuma-xyz/node-gateway/assets/6441090/36040916-a46a-4a28-afc1-abe7a39815c8)

Reply:
nice find, do we want 400 as well?  aka `>=` vs `>`?

Reply:
Good catch!